compiler/fir/raw-fir/fir-common/src/org/jetbrains/kotlin/fir/builder/Context.kt

```diff
@@ -18,7 +18,8 @@ import org.jetbrains.kotlin.name.Name
 class Context<T> {
     lateinit var packageFqName: FqName
     var className: FqName = FqName.ROOT
-    val currentClassId get() = ClassId(packageFqName, className, firFunctionTargets.isNotEmpty())
+    fun currentClassId(isLocal: Boolean = false) =
```

Comment:
It's a little odd that the `currentClassId` depends on the caller knowing whether it's currently in a "local context" or not. Is it possible to add an `isLocal` bit to `Context`? This could be set within `BaseFirBuilder.withChildClassName` every time `className` is updated; e.g., change the call to `withChildClassName(className, isLocal)`. You probably need to use a stack of `isLocal` bits, so you can correctly restore when the child is "popped".

Let me know if this makes sense or if it's not possible.

Reply:
Makes sense. Updated as per the suggestion.